src/discordtipbot/discordbot.cr

```diff
@@ -500,13 +500,14 @@ class DiscordBot
       break if new_msgs.last.timestamp < before
     end
 
-    msgs.reject!(&.author.bot)
-    msgs.reject! { |x| x.timestamp < before }
-    msgs.reject! { |x| x.author.id == msg.author.id }
-
-    authors = Array(UInt64).new
-    msgs.each { |x| authors << x.author.id }
-    authors.uniq!
-    return authors
+    authors = Set(UInt64).new
```

Comment:
I wonder if it would be worth doing [`Set#new(initial_capacity)`](https://crystal-lang.org/api/0.24.1/Set.html#new%28initial_capacity%3Dnil%29-class-method) with `msgs.size`, since it will always be less than that amount. Not sure, probably doesn't matter that much.

Reply:
Gonna merge this now. If your knowledge about this changes in the future please let me know